include/zephyr/kernel.h

```diff
@@ -6224,6 +6224,17 @@ int k_thread_runtime_stats_enable(k_tid_t thread);
  */
 int k_thread_runtime_stats_disable(k_tid_t thread);
 
+/**
+ * @brief Check if runtime statistics gathering is enabled for a thread
+ *
+ * This function checks the status of runtime statistics gathering for the
+ * specified thread and returns the current usage_lock status.
+ *
+ * @param thread ID of thread
+ * @return -EINVAL if invalid thread ID, otherwise return usage_lock status
+ */
+int k_thread_runtime_stats_is_enabled(k_tid_t thread);
```

Comment:
This is a bad design.

If you want this function to return a "status," then make it return a status (i.e. `0` on success) and nothing else -- `track_usage` should be set via a `bool *` parameter.

If you want to make this function return `bool track_usage`, then make it return `bool`.

Reply:
I don't seem to fully understand. If you're referring to the fact that it shouldn't mix error handling, which leads to a return value range of [0, 1, -1], my thought is that error handling can make it easier to catch errors when using this function. If we assume its implementation is safe, meaning tid cannot be None, I could remove the error handling, making the return value range [0, 1].